docs/src/customization.md

```diff
@@ -74,6 +74,71 @@ There is an example in the [official ST documentation](https://www.sublimetext.c
 
 The following tables give an overview about the scope names used by LSP.
 
+### Semantic Highlighting
+
+!!! info "This feature is only available if the server has the *semanticTokensProvider* capability."
+    Language servers which support semantic highlighting are for example *clangd* and *rust-analyzer*.
+
+In order to support semantic highlighting, the color scheme requires a special rule with a background color set for semantic tokens, which is (marginally) different from the original background.
+LSP automatically adds such a rule to the built-in color schemes from Sublime Text.
+If you use a custom color scheme, select `UI: Customize Color Scheme` from the Command Palette and add for example the following code:
+
+```json
+{
+    "rules": [
+        {
+            "scope": "meta.semantic-token",
+            "background": "#00000101"
+        },
+    ]
+}
+```
+
+Furthermore it is possible to adjust the colors for semantic tokens by applying a foreground color to the individual token types:
+
+| scope | [SemanticTokenTypes](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#semanticTokenTypes) |
```

Comment:
I haven't really looked deep into the implementation but should token modifiers be also mentioned? And is it possible for those to be taken into account in color schemes?

Reply:
I could mention modifiers in the docs too (later), but currently it is not possible to address them in color schemes. The only way I could imagine how it might work would be to add another separate meta scope for each modifier, and use a convention to order them alphabetically. But it would be better if we also have a way to adjust scopes for tokens with modifiers in the client configuration. The current implementation for modifiers isn't ideal, so if you have an idea for a good data structure which would allow to do that, it would be appreciated.